autosklearn/util/data.py

```diff
@@ -14,6 +25,96 @@
+    dataset_compression: Mapping[str, Any],
+    memory_limit: int
+) -> DatasetCompressionSpec:
+    """Validates and return a correct dataset_compression argument
+
+    The returned value can be safely used with `reduce_dataset_size_if_too_large`.
+
+    Parameters
+    ----------
+    dataset_compression: Mapping[str, Any]
+        The arg to validate
+
+    Returns
+    -------
+    DatasetCompressionSpec
+        The validated and correct dataset compression spec
+    """
+    if isinstance(dataset_compression, Mapping):
+        # Fill with defaults if they don't exist
+        dataset_compression = {
+            **default_dataset_compression_arg,
+            **dataset_compression
+        }
+
+        # Must contain known keys
+        if set(dataset_compression.keys()) != set(default_dataset_compression_arg.keys()):
+            raise ValueError(
+                f"Unknown key in dataset_compression, {list(dataset_compression.keys())}."
+                f"\nPossible keys are {list(default_dataset_compression_arg.keys())}"
+            )
+
+        memory_allocation = dataset_compression["memory_allocation"]
+
+        # "memory_allocation" must be float or int
+        if not (isinstance(memory_allocation, float) or isinstance(memory_allocation, int)):
+            raise ValueError("key 'memory_allocation' must be an `int` or `float`")
+
+        # "memory_allocation" must be in (0,1) if float
+        if isinstance(memory_allocation, float) and not (0.0 < memory_allocation < 1.0):
+            raise ValueError(
+                "key 'memory_allocation' if float must be in (0, 1)"
+                f"\n{dataset_compression}"
+            )
+
+        # "memory_allocation" if absolute, should be > 0 and < memory_limit
+        if isinstance(memory_allocation, int) and not (0 < memory_allocation < memory_limit):
+            raise ValueError(
+                f"key 'memory_allocation' if int must be in (0, memory_limit={memory_limit})"
+                f"\n{dataset_compression}"
+            )
+
+        # "methods" must be non-empty sequence
+        if (
+            not isinstance(dataset_compression["methods"], Sequence)
+            or len(dataset_compression["methods"]) <= 0
+        ):
+            raise ValueError(
+                "key 'methods' must be a non-empty list"
+                f"\n{dataset_compression}"
+            )
+
+        # "methods" must contain known methods
+        if any(
+            method not in cast(Sequence, default_dataset_compression_arg["methods"])  # mypy
+            for method in dataset_compression["methods"]
+        ):
+            raise ValueError(
+                f"key 'methods' can only contain {default_dataset_compression_arg['methods']}"
+                f"\n{dataset_compression}"
+            )
+
+        return cast(DatasetCompressionSpec, dataset_compression)
+    else:
+        raise ValueError(f"Unknown type for `dataset_compression` {dataset_compression}")
```

Comment:
And again here.

Reply:
For sure, although I do print out the object which is arguable a bit more useful for debugging than just a type

Reply:
Then maybe both?